### PR TITLE
kodi: update to latest

### PR DESCRIPTION
**Pull request by CvH**
- latest kodi 20
- added workaround for ?hack introduced at https://github.com/xbmc/xbmc/pull/20413
- supersedes https://github.com/LibreELEC/LibreELEC.tv/pull/5872
if a better solution is found  the workaround could be replaced, for now it looks working

build + runs for generic

### CODE REVIEW

File: packages/mediacenter/kodi/package.mk [hunk 1]
```diff
@@ -12,6 +12,12 @@ PKG_DEPENDS_TARGET="toolchain JsonSchemaBuilder:host TexturePacker:host Python3
 PKG_LONGDESC="A free and open source cross-platform media player."
 PKG_BUILD_FLAGS="+speed"
 
+post_unpack() {
+  # don't build internal TexturePacker
+  sed -i 's|set(INTERNAL_TEXTUREPACKER_INSTALLABLE TRUE)|set(INTERNAL_TEXTUREPACKER_INSTALLABLE FALSE)|' \
+    $(get_build_dir kodi)/cmake/modules/FindTexturePacker.cmake
+}
+
```
Comment (HiassofT) [inline packages/mediacenter/kodi/package.mk:20]: IMO it'd be better to handle this via a patch.
eg kodi-100.26-disable-internal-texturepacker.patch
```
--- a/cmake/modules/FindTexturePacker.cmake
+++ b/cmake/modules/FindTexturePacker.cmake
@@ -52,10 +52,8 @@ if(NOT TARGET TexturePacker::TexturePacker::Executable)
       endif()
     endif()
 
-    # Ship TexturePacker only on Linux and FreeBSD
-    if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" OR CMAKE_SYSTEM_NAME STREQUAL "Linux")
-      set(INTERNAL_TEXTUREPACKER_INSTALLABLE TRUE)
-    endif()
+    # Don't install TexturePacker
+    set(INTERNAL_TEXTUREPACKER_INSTALLABLE FALSE)
 
     # Use it during build if build architecture is same as host
     # (not cross-compiling) and TEXTUREPACKER_EXECUTABLE is not found
```

Comment (vpeter4) [reply packages/mediacenter/kodi/package.mk:20]: No need to set INTERNAL_TEXTUREPACKER_INSTALLABLE to FALSE.

**Comment by CvH** [comment]
pushed changes, just added the removal like @vpeter4 advised

**Comment by lrusak** [comment]
have you tried just setting `-D INTERNAL_TEXTUREPACKER_INSTALLABLE=OFF` ??

**Comment by HiassofT** [comment]
@lrusak yes, I already tried that and it didn't work - the set in cmake file takes precedence over that

**Review by HiassofT — APPROVED** [review]

**Comment by HiassofT** [comment]
Runtime test on RPi4 was fine.

If there are better solutions to the texturepacker issue they can be implemented later, for now I'm happy with the current workaround

**Pull request merged**